src/Quizzer.API.BL.UnitTests/QuestionFacadeTest.cs

```diff
@@ -0,0 +1,188 @@
+                Answers = {
+                    new AnswerDetailModel {
+                        Id = Guid.NewGuid(),
+                        Text = "some text"
+                    },
+                    new AnswerDetailModel {
+                        Id = Guid.NewGuid(),
+                        Text = "some text",
+                        Type = DAL.Enums.AnswerFormat.OrderChoice
+                    },
+                    new AnswerDetailModel {
+                        Id = Guid.NewGuid(),
+                        Text = "some text",
+                        Type = DAL.Enums.AnswerFormat.MultiChoice
+                    }
+                }
+            };
+
+            _mapperMock.Setup(mapper => mapper.Map<QuestionDetailModel>(It.IsAny<QuestionDetailModel>())).Returns(questionDetailModel);
+
+            // Act 
+            questionFacade.GetDetailById(questionDetailModel.Id);
+
+            // Assert
+            Assert.Equal(3, questionDetailModel.Answers.Count);
+            foreach (AnswerDetailModel answerModel in questionDetailModel.Answers)
+                Assert.Equal(questionDetailModel.QuestionType, answerModel.Type);
+        }
+
+        [Fact]
+        public void AddAnswer_With_Duplicate_Id_Throws_Exception() {
+            // Arrange
+            QuestionFacade questionFacade = GetFacadeWithForbiddenDependencyCalls();
+            var dupliciteAnswerId = Guid.NewGuid();
+            var questionId = Guid.NewGuid();
+
+            Mock<IRepository<AnswerEntity>> answerRepositoryMock = new(MockBehavior.Strict);
+            _uowMock.Setup(uow => uow.GetRepository<AnswerEntity>()).Returns(answerRepositoryMock.Object);
+            answerRepositoryMock.Setup(rep => rep.Dispose());
+
+            _questionRepositoryMock
+                .Setup(rep => rep.GetById(It.IsAny<Guid>(), nameof(QuestionEntity.Answers)))
+                .Returns(new QuestionEntity {
+                    Id = questionId,
+                    Text = "Text of question",
+                    Type = DAL.Enums.AnswerFormat.SingleChoice,
+                    Answers = {
+                            new AnswerEntity {Id = Guid.NewGuid(), IsCorrect = false, Text = "aw1"},
+                            new AnswerEntity {Id = Guid.NewGuid(), IsCorrect = false, Text = "aw2"},
+                            new AnswerEntity {Id = dupliciteAnswerId, IsCorrect = true, Text = "aw3"},
+                            new AnswerEntity {Id = Guid.NewGuid(), IsCorrect = false, Text = "aw4"}
+                    }
+                });
+
+            // Act & Assertl
+            Assert.Throws<ArgumentException>(() => questionFacade.AddAnswer(questionId, dupliciteAnswerId));
+        }
+
+        [Theory]
+        [InlineData(1, 2, 3, 1)]
+        [InlineData(1, 2, 3, 2)]
+        [InlineData(1, 2, 3, 3)]
+        public void AddAnswer_With_Bad_Order_Throws_Exception(int order1, int order2, int order3, int newOrder) {
```

Comment:
I also ment using "_" only to separate Action_Conditions_Expected in name of function, like:
`AddAnswer_WithBadOrder_ThrowsException`, it would be more readable imo.

Reply:
fixed

Reply:
I ment mergin every word in a section: 
`Save_Answer_With_Bad_Order_ThrowsException  `-> `SaveAnswer_WithBadOrder_ThrowsException `.
But I see they have different naming convention in CB, so I guess it ok.